### DISCUSSION TITLE
Using reorderable() with spaties eloquent-sortable package works great, until you use the spatie package 'grouping' functionality

### DISCUSSION THREAD

**Discussion by SimonMacIntyre** [filamentphp/filament#8803] [Ideas]
In the Filament docs, it mentions using Spatie's eloquent-sortable package in combination with filaments `reorderable('order')` pattern.

It works great~ unless you are also using this feature of the spatie package:

```php
public function buildSortQuery()
    {
        return static::query()->where('topic_group_id', $this->topic_group_id);
    }
``` 

It would be good if the Filament reorder drag and drop functionality respected this 'group by', and allowed you to `reorder within the group`, in the presented view.  Notice how in my 2 uploaded images, that the concept of `groups` goes completely out the window when trying to reorder. 

This basically makes `reorder` incompatible with the grouping, since I would never allow this UX, even if the reorder was functional under this UX. 

<img width="1241" alt="image" src="https://github.com/filamentphp/filament/assets/1260938/d9574067-e58b-4bf5-9f77-eab12c793723">

<img width="1234" alt="image" src="https://github.com/filamentphp/filament/assets/1260938/e422a4fd-8133-4831-8df5-a84d18ee58da">

I wouldn't call this a bug, but it is unfortunate since my only choice now is to essentially not use grouping, because reordering is a hard requirement here, and is a big loss to lose the `grouping` view in the table. 

If I'm missing a way to workaround this, or missing some way to achieve what I am talking about, please do let me know! I am new to Filament! 

LOVE filament so far!

**Comment by binaryfire** [comment]
Hey Simon. Did you end up finding a workaround for this?

**Reply by SimonMacIntyre** [reply]
Kind of -- I eventually found out that if I implement a `filter` of the 'group by' that I'm using that the drag and drop works. It just requires ensuring any admin users are aware that using the sort function WITHOUT applying the appropriate filter is gonna be a bad time. 

But with the filter on it works great. 

For example I have a filter which basically is `where some_column = some_foreign_key_id` so they can then sort that subset in that view.

**Reply by binaryfire** [reply]
Ah ok, interesting. The panel I’m working is for end users so unfortunately I can’t use that approach. Hopefully there will be a better solution down the road.
